pagopa/pn-frontend
# PN-2269-fe-validazione-campi-di-input-in-invio-manuale-di-una-notifica

## Short description
I added validations for the firstName and taxId fields in the Recipient component, the second step in the workflow to create a notification. I also fixed some details regarding the radio buttons for "Persona Fisica" and "Persona Giuridica' in the same component.

## List of changes proposed in this pull request
- The complete denomination of the recipient for a PF (firstName + " " + lastName) should not be longer than 80 characters, hence the sum of the lengths of firstName and lastName shouldn't exceed 79
- Tax ID should accept both 16-char and 11-digit formats for PG, only 16-char format for PF.

## How to test
Enter PA, in the Notifiche page click the button "Invia una nuova notifica", fill the first step about "Informazioni Preliminari", you should arrive to the step about "Destinatario".   
In that page you can check that the FE detect an error and disable the "Continua" button, if either the combined lengths of "Nome" and "Cognome" is more that 79, or else the "Codice Fiscale Destinatario" does not conform with the 16-char format for a CodiceFiscale.  
At the moment the "Persona Giuridica" option is disabled. If you're running the FE locally you can enable it by just finding "disabled" in `Recipient.tsx`. If you allow the selection of this option, by selecting it you should notice the following: 
1. The value of "Nome" and "Cognome", or "Ragione Sociale", gets cleaned when you change from "Persona Fisica" to "Persona Giuridica" or viceversa.
2. The length of "Ragione Sociale" should not be greater than 80
3. If "Persona Giuridica" is selected, both the 16-char and the 11-digit formats for Codice Fiscale should be acceptable.
